marlowe-dashboard-client/src/Contract/State.purs

```diff
@@ -79,12 +80,34 @@ dummyState =
 
   emptyPlutusAppId = PlutusAppId UUID.emptyUUID
 
-  emptyMarloweParams = { rolePayoutValidatorHash: mempty, rolesCurrency: CurrencySymbol { unCurrencySymbol: "" } }
-
   emptyMarloweData = { marloweContract: contract, marloweState: emptyMarloweState }
 
   emptyMarloweState = Semantic.State { accounts: mempty, choices: mempty, boundValues: mempty, minSlot: zero }
 
+-- this is for making a placeholder state for the user who created the contract, used for displaying
+-- something before we get the MarloweParams back from the WalletCompanion app
+mkPlaceholderState :: PlutusAppId -> MetaData -> Contract -> State
+mkPlaceholderState followerAppId metaData contract =
+  { tab: Tasks
+  , executionState: initExecution zero contract
+  , pendingTransaction: Nothing
+  , previousSteps: mempty
+  , mMarloweParams: Nothing
+  , followerAppId
+  , selectedStep: 0
+  , metadata: metaData
+  -- Note we filter out PK parties here. This is because we don't have a design for displaying
+  -- them anywhere, and because we are currently only using one in a special case (in the Escrow
+  -- with Collateral contract), where it doesn't make much sense to show it to the user anyway.
+  -- If we ever want to use PK parties for other purposes, we will need to rethink this.
+  , participants: Map.fromFoldable $ map (\x -> x /\ Nothing) (getRoleParties contract)
```

Comment:
we could join this expression and comment into a function and call it from mkInitialState and mkPlaceholderState